Sources/PostgresNIO/New/Data/Optional+PSQLCodable.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  File.swift
+//  
+//
+//  Created by Fabian Fett on 13.01.21.
+//
+
+extension Optional: PSQLDecodable where Wrapped: PSQLDecodable {
+    static func decode(from byteBuffer: inout ByteBuffer, type: PSQLDataType, context: PSQLDecodingContext) throws -> Optional<Wrapped> {
+        preconditionFailure("This code path should never be hit.")
```

Comment:
I don't know, I've seen `Codable` do some wild things with optionals, especially when people try to get clever with their decoding.

Reply:
I'm sorry, I'm not sure if I can follow you here?

Reply:
You say that this code path should never be hit, but I've found `Codable` to be untrustworthy when it comes to which sequence of container methods gets invoked for `Optional`s. To be fair, you're probably correct in this case.